src/main/java/igrad/model/course/CourseInfo.java

```diff
@@ -47,12 +47,63 @@ public CourseInfo(Optional<Name> name, Optional<Cap> cap) {
      * requirement list.
      */
     public static Cap computeCap(List<Module> moduleList) {
-        double cap = 4.5;
-        /**
-         * TODO: Teri, please fill the details. Iterate through all modules in the module list and
-         * for each module with a grade, add them together to calcualte the cap
-         * Lemme know if you find this too hard, i can do it :)
-         */
+        double cap = 0;
```

Comment:
good but can be improved; package all this logic into the `model/modules/Grade.java` class